### DISCUSSION TITLE
DNA Database of Disease-Causing genes

### DISCUSSION THREAD

**Discussion by VerisimilitudeX** [VerisimilitudeX/DNAnalyzer#246] [Ideas]
From @LimesKey:

Create a database with common DNA Gene strings known to cause medical illnesses and conditions, sprinkle some machine learning combined with randomness percentage to account for genetic drift and evolution over time.

To increase accuracy with the reference database DNA is to have lots of metadata with that gene string, such as the average age of the person with that mutated gene string and with correlation with other genes on the person's DNA.

**Comment by LimesKey** [comment]
This is a very large topic and feature, later today i'm going to split this feature into multiple smaller, easier to understand issues. I feel like most of the people on Github that are coders don't have a very good biology background. @Verisimilitude11

**Comment by VerisimilitudeX** [comment]
Sounds good @LimesKey, I agree with that.

**Comment by VerisimilitudeX** [comment]
@LimesKey, once https://github.com/Verisimilitude11/DNAnalyzer/issues/205 gets resolved, I feel like the knowledge-gap issue will get slightly resolved. However, splitting this feature into bite-sized issues would also be helpful, so feel free to proceed. I'm going to convert this issue into a discussion thread. Thanks again!

**Reply by LimesKey** [reply]
@Verisimilitude11 Right now I'm working on trimming a sequenced FASTA file when given to the program, more than half of the information in the file the program may not use. It should significantly increase speed and help with the out-of-memory error message. I also want this program to work best for genotyped files, but I've hit a roadblock as there's not a lot of information I could find on this.

**Reply by VerisimilitudeX** [reply]
Got it. Are you talking about the headers (for example, chromosome and arms)?

**Reply by LimesKey** [reply]
I think so, yes. I wonder how a company like 23andMe does it, do they only collect specific chromoesomes or parts or do they collect all the genetic information and use a program to cut some out? If they only collect specific DNA parts like chromosomes, which chromosomes do they collect?

**Reply by VerisimilitudeX** [reply]
I'm pretty sure that they collect all parts of the genetic information and they separate it out into the 22 chromosomes (in addition to X & Y). They're able to do so because the chromosomes have different sizes, banding patterns, and centromere patterns. You might find this helpful: https://learn.genetics.utah.edu/content/basics/readchromosomes/#:~:text=To%20%22read%22%20a%20set%20of%20chromosomes%2C%20scientists%20use,daughter%20cells%20during%20cell%20division%20%28mitosis%20and%20meiosis%29.

**Reply by LimesKey** [reply]
That seems expensive, I thought that company uses Genotyping because of the cost of sequencing.

**Reply by VerisimilitudeX** [reply]
Well, I'm not sure about companies, but that's what colleges/universities/hospitals use since it's more accurate. It may be that companies use genotyping, I'm not completely sure about that.

**Comment by i2kmt** [comment]
Hey, this seems really interesting and I would like to contribute. I can help with making a list of different genes in categories, having in mind that you would like to see an evolutionary background. But as @LimesKey said, it is indeed a very large topic, including the complexity and variety of different cellular mechanisms that could be affected by a gene, causing a disease.

**Reply by VerisimilitudeX** [reply]
Great, thank you for your offer to contribute! Your suggestion of categorizing genes based on their evolutionary background is a great idea and could provide valuable insight into how certain genes have evolved to play specific roles in disease development. We appreciate your understanding of the complexity and diversity of cellular mechanisms that can be affected by genes.

To get started, we would suggest forking the GitHub repository specifically for this task, and then making a list of genes and their corresponding categories. As you work on the list, feel free to share your progress and ask any questions you may have. Our team will be happy to assist you along the way (feel free to join our Discord server here: https://discord.gg/X3YCvGf2Ug). Thank you again for your interest in contributing to this project!

**Reply by i2kmt** [reply]
Great, thanks! I joined the server. Will be asking my questions there, as I go along :)

**Reply by VerisimilitudeX** [reply]
Sounds good! I'm [@Verisimilitude](https://github.com/Verisimilitude11) in the server and you can ask [@LimesKey](https://github.com/LimesKey) as well.
